### PR TITLE
User profile component added

### LINKED ISSUE #26: Add Profile Page for User UI
API : /user/details (GET)

### PR DESCRIPTION
**Pull request by DebugAgrawal**
Resolves #26 

- User Profile card
- Users posts

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "web-vitals": "^1.1.1"
   },
   "scripts": {
-    "start": "NODE_ENV=development craco start",
+    "start": "craco start",
```
Comment (bodhisha) [inline frontend/package.json]: You need not have to push this.

File: frontend/src/constants.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const ENV = process.env.NODE_ENV;
+const ENV = "development";
```
Comment (bodhisha) [inline frontend/src/constants.js]: This too..

**Review by bodhisha — CHANGES_REQUESTED** [review]
Looks Good.. 🎉 
I think you need to remove the changes in the package.json, and constants file

File: frontend/package.json (hunk 1 above)
Comment (DebugAgrawal) [reply frontend/package.json]: > Done

File: frontend/src/constants.js (hunk 2 above)
Comment (DebugAgrawal) [reply frontend/src/constants.js]: > Done

File: frontend/src/Components/UserProfile.js [hunk 3]
```diff
@@ -1,13 +1,79 @@
-import React, { useContext } from "react";
+import React, { useEffect, useState, useContext } from "react";
 import { AuthContext } from "../Context/AuthContext";
+import Post from "./Post/Post";
+import axios from "axios";
+import { Loading } from "../Components/Common/Loader";
+import { POST_GET_ALL_URL } from "../constants";
 
-export default function UserFeed() {
+export default function UserProfiele() {
```
Comment (bodhisha) [inline frontend/src/Components/UserProfile.js]: Small Typo.. :p `UserProfile` @DebugAgrawal

**Review by bodhisha — CHANGES_REQUESTED** [review]

File: frontend/src/Components/UserProfile.js (hunk 3 above)
Comment (DebugAgrawal) [reply frontend/src/Components/UserProfile.js]: oh lol 😂

**Review by bodhisha — APPROVED** [review]

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "lockfileVersion": 1
+}
```
Comment (aitikgupta) [inline package-lock.json]: Why do we need this?

File: frontend/src/Components/UserProfile.js [hunk 5]
```diff
@@ -1,13 +1,79 @@
...
+                <p className="text-base font-medium text-black-700 mr-2 box-content text-xl font-mono ">
+                  {user[0].description}
+                </p>
+              </div>
+            </div>
+
+            <div className="text-left text-center w-1/2 mx-auto">
+              <button className="bg-transparent hover:bg-blue-500 text-gray-700 font-semibold hover:text-white py-2 px-6 border border-gray-600 hover:border-transparent rounded">
+                Edit Profile
+              </button>
```
Comment (aitikgupta) [inline frontend/src/Components/UserProfile.js:62]: But Edit Profile is not doing anything?

File: frontend/src/Components/UserProfile.js [hunk 6]
```diff
@@ -1,13 +1,79 @@
...
+export default function UserProfile() {
   const { user, token } = useContext(AuthContext);
+  const [posts, setPosts] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    axios
+      .get(POST_GET_ALL_URL, {
+        headers: {
```
Comment (aitikgupta) [inline frontend/src/Components/UserProfile.js:17]: You're using the wrong Axios URL. You need to either use `/user/current_user` or (`/user/details` with user_id).

This is calling the _whole_ Posts URL.

**Review by aitikgupta — CHANGES_REQUESTED** [review]
This needs more work with the URL end

File: frontend/src/Components/UserProfile.js (hunk 6 above)
Comment (bodhisha) [reply frontend/src/Components/UserProfile.js:17]: My PR will fix this.. I'm using the other API.

File: frontend/src/Components/UserProfile.js (hunk 5 above)
Comment (DebugAgrawal) [reply frontend/src/Components/UserProfile.js:62]: currently it's static but we will be needing that while applying the edit profile/bio logic

Comment (aitikgupta) [reply frontend/src/Components/UserProfile.js:62]: Yep, I've added an issue https://github.com/Open-Sourced-Olaf/SocioMark/issues/84 for the same

**Review by aitikgupta — APPROVED** [review]
Looks good now 👍🏼

**Pull request merged**